specs/PushNotifications/ToastNotifications-spec.md

```diff
@@ -0,0 +1,571 @@
+    enum ToastNotificationSetting
+    {
+        Enabled, // Toast is not blocked by settings or group policy
+        DisabledForApplication, // Toast is blocked by a user defined App Setting
+        DisabledForUser, // Toast is blocked by a user defined Global Setting
+        DisabledByGroupPolicy, // Toast is blocked by Group Policy
+        DisabledByManifest, // Toast is blocked by a setting in the manifest. Only for packaged applications.
+    };
+
+    [experimental]
+    [flags]
+    // Some basic predefined Toast Payload Templates
+    enum ToastTemplateType
+    {
+        ToastImageAndText01,
+        ToastImageAndText02,
+        ToastImageAndText03,
+        ToastImageAndText04,
+        ToastText01,
+        ToastText02,
+        ToastText03,
+        ToastText04,
+    };
+
+    [experimental]
+    [flags]
+    // The Result for a Toast Progress related operation
+    enum ToastProgressResult
+    {
+        Succeeded, // The progress operation succeeded
+        Failed, // The progress operation failed
+        NotificationNotFound, // The progress operation failed to find a toast to process updates
+    };
+
+    [experimental]
+    [flags]
+    enum ToastPriority
+    {
+        Default, // The notification should have default behavior in terms of delivery and display priority during connected standby mode.
+        High, // The notification should be treated as high priority. For desktop PCs, this means during connected standby mode the incoming notification can turn on the screen for Surface-like devices if it doesn't have a closed lid detected.
+    };
+
+    [experimental]
+    // Represent a toast Notification construct
+    runtimeclass ToastNotification
+    {
+        // Initialize a new Toast using an XML Payload.
+        ToastNotification(Windows.Data.Xml.Dom.XmlDocument payload);
+
+        // Unique identifier used to replace a notification within a group.
+        String Tag;
+
+        // Unique identifier for a toast group in the app
+        String Group;
+
+        // A unique identifier for the Toast generated by the platform.
+        UInt32 ToastIdentifier;
+
+        // The notification Xml Payload
+        Windows.Data.Xml.Dom.XmlDocument Payload{ get; };
+
+        // Gets or sets additional information about the toast progress.
+        ToastProgressData ProgressData;
+
+        // Gets or sets the time after which a toast notification should not be displayed.
+        Windows.Foundation.DateTime ExpirationTime;
+
+        // Indicates whether the toast will remain in the Action Center after a reboot.
+        Boolean ExpiresOnReboot;
+
+        // Gets or sets the priority for a Toast.
+        // Hints on how and at what urgency level a notification should be presented to the user (whether to wake up the screen, etc).
+        ToastPriority Priority;
+
+        // Gets or sets whether a toast's pop-up UI is displayed on the user's screen.
+        Boolean SuppressPopup;
```

Comment:
Or perhaps a bit more clearer `IsVisible`?

Reply:
Going with SuppressDisplay for now since this is technically still visible in the actionCentre even after the value is set to true. It's just that the Popup won't show at the bottom right corner of the screen and its a silent action centre update.

Reply:
Oh, I didn't realize that SuppressDisplay=true still showed it in action center. I thought it suppressed it from displaying anywhere, like a raw notification.

Boolean ShowAsPopup; // default true

Reply:
No. The popup won't show if it is set to false and the screen will still be turned off